sdk/core/core-client-rest/src/common.ts

```diff
@@ -48,8 +53,132 @@ export type RequestParameters = {
    * Path parameters for custom the base url
    */
   pathParameters?: Record<string, any>;
+
+  /**
+   * The number of milliseconds a request can take before automatically being terminated.
+   */
+  timeout?: number;
+
+  /**
+   * Callback which fires upon upload progress.
+   */
+  onUploadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * Callback which fires upon download progress.
+   */
+  onDownloadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * The signal which can be used to abort requests.
+   */
+  abortSignal?: AbortSignalLike;
+
+  /**
+   * Options used when tracing is enabled.
+   */
+  tracingOptions?: OperationTracingOptions;
+
+  /**
+   * A function to be called each time a response is received from the server
+   * while performing the requested operation.
+   * May be called multiple times.
+   */
+  onResponse?: RawResponseCallback;
 };
 
+/**
+ * A function to be called each time a response is received from the server
+ * while performing the requested operation.
+ * May be called multiple times.
+ */
+export type RawResponseCallback = (
+  rawResponse: FullOperationResponse,
+  error?: unknown
+) => void;
```

Comment:
The original RawResponseCallback also take a parameter called flatResponse which is the result after deserialization, as we are deserialize the response in the api layer now, I don't think we should add it into the @azure-rest/core-client, so I just removed this parameters.

Reply:
Agreed!